ocean-client/reinvest-cloudtemplate.json

```diff
@@ -0,0 +1,404 @@
+                ]
+              }
+            ]
+          },
+          "AssertDescription": "Reinvest can not be empty"
+        }
+      ]
+    }
+  },
+  "Conditions": {
+    "CreateTelegramLogs": {
+      "Fn::And": [
+        {
+          "Fn::Not": [
+            {
+              "Fn::Equals": [
+                {
+                  "Ref": "TelegramLogsChatId"
+                },
+                ""
+              ]
+            }
+          ]
+        },
+        {
+          "Fn::Not": [
+            {
+              "Fn::Equals": [
+                {
+                  "Ref": "TelegramLogsToken"
+                },
+                ""
+              ]
+            }
+          ]
+        }
+      ]
+    },
+    "CreateTelegramNotifications": {
+      "Fn::And": [
+        {
+          "Fn::Not": [
+            {
+              "Fn::Equals": [
+                {
+                  "Ref": "TelegramNotificationChatId"
+                },
+                ""
+              ]
+            }
+          ]
+        },
+        {
+          "Fn::Not": [
+            {
+              "Fn::Equals": [
+                {
+                  "Ref": "TelegramNotificationToken"
+                },
+                ""
+              ]
+            }
+          ]
+        }
+      ]
+    }
+  },
+  "Resources": {
+    "reinvestRole": {
+      "Type": "AWS::IAM::Role",
+      "Properties": {
+        "AssumeRolePolicyDocument": {
+          "Version": "2012-10-17",
+          "Statement": [
+            {
+              "Effect": "Allow",
+              "Principal": {
+                "Service": [
+                  "lambda.amazonaws.com"
+                ]
+              },
+              "Action": [
+                "sts:AssumeRole"
+              ]
+            }
+          ]
+        },
+        "Path": "/",
+        "Policies": [
+          {
+            "PolicyName": "LogsAndSSM",
+            "PolicyDocument": {
+              "Version": "2012-10-17",
+              "Statement": [
+                {
+                  "Effect": "Allow",
+                  "Action": [
+                    "logs:CreateLogGroup",
+                    "logs:CreateLogStream",
+                    "logs:PutLogEvents"
+                  ],
+                  "Resource": "*"
+                },
+                {
+                  "Effect": "Allow",
+                  "Action": [
+                    "ssm:Describe*",
+                    "ssm:Get*",
+                    "ssm:List*"
+                  ],
+                  "Resource": {
+                    "Fn::Sub": "arn:aws:ssm:${AWS::Region}:${AWS::AccountId}:parameter/defichain-maxi/*"
+                  }
+                }
+              ]
+            }
+          }
+        ]
+      }
+    },
+    "reinvestTrigger": {
+      "Type": "AWS::Events::Rule",
+      "Properties": {
+        "Name": "reinvestTrigger",
+        "Description": "Triggers the reinvest",
+        "ScheduleExpression": "rate(53 minutes)",
```

Comment:
why every 53 minutes? to avoid being executed in sync with vault maxi?

Reply:
prime number, roughly every hour. but this way its very unlikely to trigger at the same time as maxi.